dev_environment/setup_dev_env.sh

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# this script need to run as root account, check it
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root"
+    exit 1
+fi
+
+# Get cpu architecture, arm or amd
+ARCH=$(uname -p)
+if [[ "$ARCH" = "arm" ||  "$ARCH" = "aarch64" ]];
+then
+  echo "Setting l3af dev environment for arm"
+  arch=arm64
+elif [[ "$ARCH" = "x86_64" || "$ARCH" = "i386" ]];
+then
+  echo "Setting l3af dev environment for amd64"
+  arch=amd64
```

Comment:
For Azure VM, uname -p return x86_64 instead of i386, no further check needed.  If the MAC return i386 from the commands of uname -p or arch, I don't think we need uname -m to check the kernel 32 bit/64bit anymore.  IOS has been 64bit for long time.  I don't think ebpf supports 32 bit kernel anyway (only classical bpf supported 32bit kernel).

Reply:
Since go has both x86_64 and 386 distribution, I think we should set properly based on uname -m

Reply:
Add the handle for uname -m if i386 detected during uname -p